x-pack/plugins/security_solution/cypress/tasks/login.ts

```diff
@@ -42,6 +44,88 @@ const ELASTICSEARCH_PASSWORD = 'ELASTICSEARCH_PASSWORD';
  */
 const LOGIN_API_ENDPOINT = '/internal/security/login';
 
+// these roles are derived from https://docs.google.com/spreadsheets/d/1SGjXVS3gQyRxCN1MF2Pq3pgos3vtfpROqTvi4z-0WiU/edit#gid=0
+export const ROLES = {
+  t1_analyst: 't1_analyst',
+  t2_analyst: 't2_analyst',
+  hunter: 'hunter',
+  rule_author: 'rule_author',
+  soc_manager: 'soc_manager',
+  detections_admin: 'detections_admin',
+  platform_engineer: 'platform_engineer',
+};
+
+/**
+ * cy.visit will default to the baseUrl which uses the default kibana test user
+ * This function will override that functionality in cy.visit by building the baseUrl
+ * directly from the environment variables set up in x-pack/test/security_solution_cypress/runner.ts
+ *
+ * @param role string role/user to log in with
+ * @param route string route to visit
+ */
+export const getUrlWithRoute = (role: string, route: string) => {
+  const theUrl = `${Url.format({
+    auth: `${role}:changeme`,
+    username: role,
+    password: 'changeme',
+    protocol: Cypress.env('protocol'),
+    hostname: Cypress.env('hostname'),
+    port: Cypress.env('configport'),
+  } as UrlObject)}${route.startsWith('/') ? '' : '/'}${route}`;
+  cy.log(`origin: ${theUrl}`);
+  return theUrl;
+};
+
+export const postRoleAndUser = (role: string) => {
+  const env = {
+    ELASTICSEARCH_URL: Cypress.env('ELASTICSEARCH_URL'),
+    ELASTICSEARCH_USERNAME: Cypress.env('ELASTICSEARCH_USERNAME'),
+    ELASTICSEARCH_PASSWORD: Cypress.env('ELASTICSEARCH_PASSWORD'),
+    KIBANA_URL: Cypress.env('KIBANA_URL'),
+  };
+  const detectionsRoleScriptPath = `./server/lib/detection_engine/scripts/roles_users/${role}/post_detections_role.sh`;
+  const detectionsRoleJsonPath = `./server/lib/detection_engine/scripts/roles_users/${role}/detections_role.json`;
+  const detectionsUserScriptPath = `./server/lib/detection_engine/scripts/roles_users/${role}/post_detections_user.sh`;
+  const detectionsUserJsonPath = `./server/lib/detection_engine/scripts/roles_users/${role}/detections_user.json`;
+
+  // post the role
+  cy.exec(`bash ${detectionsRoleScriptPath} ${detectionsRoleJsonPath}`, {
+    env,
+  });
+
+  // post the user associated with the role to elasticsearch
+  cy.exec(`bash ${detectionsUserScriptPath} ${detectionsUserJsonPath}`, {
+    env,
+  });
```

Comment:
So looks like these will get executed with each call to `loginAndWaitForPageWithoutDateRange()`, even if the user already exists, and so we'll just ignore the error and move along? Just thinking through use cases when testing the same roles/user within the same spec. Actually, is there any teardown between specs that would remove these users?

Reply:
When I was testing these scripts out the unload scripts were performing the role + user teardown https://github.com/elastic/kibana/blob/3ba7758a4fad65d0df62eafb4ffd03084195eed0/x-pack/plugins/security_solution/cypress/tasks/es_archiver.ts#L23 I could add an explicit "deleteUserAndRole" function if you think that would be helpful?

Reply:
Ahhhhh, interesting! I hadn't realized the unload handled the tear down as well. In that case, we're probably fine for most test cases since we're un/reloading data between, but wouldn't hurt to have a `deleteUserAndRole` function available in cases where we're testing without the underlying dataset changing.